cpp/oneapi/dal/algo/kmeans/detail/infer_ops_dpc.cpp

```diff
@@ -42,10 +42,13 @@ struct infer_ops_dispatcher<Policy, Float, Method, Task> {
     template struct ONEDAL_EXPORT                                         \
         infer_ops_dispatcher<dal::detail::spmd_data_parallel_policy, F, M, T>;
 
+#define INSTANTIATE_NON_DISTRIBUTED(F, M, T) \
```

Comment:
is it necessary? I thought that https://github.com/oneapi-src/oneDAL/pull/2959/files#diff-d63b2d18fea17253b19beac3bd26f5d07f2adc4e8e683a0a195b3dd8f1bf81e7R39 is enough

Reply:
It is necessary in order not to instantiate the variants of the descriptor that do not make sense (sparse + spmd).

Reply:
Looks like this may be causing a CI fail in spmd.